django/core/checks/settings_deprecation.py

```diff
@@ -0,0 +1,87 @@
+from django.conf import settings
+from django.core import checks
+
+
+def deprecated_settings_variables(*args, **kwargs):
+    """
+    This check warns users who still use deprecated settings variables.
+    """
+
+    deprecated_settings_list = (
+        # todo: add older settings
+        # Django 1.4
+        "TRANSACTIONS_MANAGED",
+        # Django 1.5
+        "SITE_ID",
+        "AUTH_PROFILE_MODULE",
+        "CSRF_COOKIE_PATH",
+        # Django 1.6
+        "DATABASE_ENGINE",
+        # Django 1.7
+        "SOUTH_DATABASE_ADAPTER",
+        "SOUTH_DATABASE_ADAPTERS",
+        "SOUTH_AUTO_FREEZE_APP",
+        "SOUTH_TESTS_MIGRATE",
+        "SOUTH_LOGGING_ON",
+        "SOUTH_LOGGING_FILE",
+        "SOUTH_MIGRATION_MODULES",
+        "SOUTH_USE_PYC",
+        # Django 1.8
+        "SEND_BROKEN_LINK_EMAILS",
+        "CACHE_MIDDLEWARE_ANONYMOUS_ONLY",
+        # Django 1.9
+        "FILE_UPLOAD_HANDLERS",
+        # Django 1.10
+        "ALLOWED_INCLUDE_ROOTS",
+        "LOGOUT_URL"
+        "SECURE_PROXY_SSL_HEADER",
+        "TEMPLATE_CONTEXT_PROCESSORS",
+        "TEMPLATE_DEBUG",
+        "TEMPLATE_DIRS",
+        "TEMPLATE_LOADERS",
+        "TEMPLATE_STRING_IF_INVALID",
+        # Django 2.1
+        "USE_ETAGS",
+        "SECURE_BROWSER_XSS_FILTER",
+        # Django 3.0
+        "DEFAULT_CONTENT_TYPE",
+        "PASSWORD_RESET_TIMEOUT_DAYS",
+        # Django 3.1
+        "DEFAULT_FILE_STORAGE",
+        "FILE_CHARSET",
+        # Django 4.0
+        "DEFAULT_HASHING_ALGORITHM",
+        "PASSWORD_RESET_TIMEOUT_DAYS",
+        "SECURE_BROWSER_XSS_FILTER",
+        # Django 4.1
+        "DEFAULT_STORAGE_CLASS",
+        # Django 4.2
+        "MIDDLEWARE_CLASSES",
+        "FILE_UPLOAD_HANDLERS",
+        "ADMINS",
+        "MANAGERS",
```

Comment:
Are those two settings really deprecated? I cannot find anything about it.

Reply:
To reiterate, i didn't check all settings by hand, just those where I immediately thought that something's off.

Also, for the next time: Please mention when something was generated by AI, because then I as a human can avoid spending time checking it.

Reply:
Sorry about that! I have a little hard time with the three places where info about this ticket goes (trac, GitHub issue, Django forum) and while creating this issue I felt that probably someone wouldn't be in the loop 😕 

This being said: I didn't dump the AI output here blindly. I checked the values and merged them with Adams thoughts, though I didn't check them well enough it seems. But since I wanted to get this thing off the ground and hopefully into Django 5.2, I needed a starting point. Since many people might have opinions on how to approach this (that's a good thing!) I didn't want to spend too much time for something that maybe gets rejected in a glimpse.

This being said, sorry for creating some extra work for you but I hope you understand my decision drivers.

Reply:
Thank you, and don't worry. I just had to vent a little! I think the change is worthwile and I'm definitely looking forward to it.

Reply:
All good, thanks for the support! ❤️ 

---

While we're at it... do you have an idea why the override settings doesnt work in the test I wrote? Even setting the settings variable directly doesn't have any effect.

Reply:
I'm not sure! Just looking at the code makes me think it should work.

That being said, maybe you could iterate through the list of deprecated settings and use `is_overridden` instead to determine whether users have set a setting themselves? https://github.com/django/django/blob/c334c1a8ff4579cdb1dd77cce8da747070ac9fc4/django/conf/__init__.py#L207-L208 -- it seems like the (internal) way to determine whether a setting is being used or not.

Reply:
Django checks for settings attributes with .isupper(). I added this. Apart from this, I'm importing the django.conf settings. Do you think we should use this class you've linked so we can utilize this method? Otherwise, it seems to work now as expected.

Reply:
Resolving this since I didn't get any complains anymore.